### PR TITLE
Feat: while statement

### PR DESCRIPTION
**Pull request by GangstersPairOfDice**
Added support for while statement using previously implemented syntax, new token, etc.

Usage in main.pych:
while ( expression ) { print("jeff"); }

### CODE REVIEW

File: src/pychart/__init__.py [hunk 1]
```diff
@@ -1 +1 @@
-__version__ = '0.0.0'
+__version__ = '0.6.9'
```
Comment (miningape) [inline src/pychart/__init__.py:1]: Classy

File: src/pychart/_interpreter/pyparser.py [hunk 2]
```diff
@@ -56,13 +56,33 @@ def statement(self) -> Stmt:
...
+        self.consume( TokenType.RIGHT_PEREN, 'Expected ")" after expression in WHILE statement' )
+
+        while_body = self.statement()
+
+        #saigo ni..........Baba jonhson Bizzaeraaeh...........
+        # so why is this funny ( to m e )
+        # sounds like sterioetypical anime voice
+        # good vocal tonage ( sounds niceto say) good vowels, mixed with with good constancants
+        # reminds me of mitsubishi materieals vine
+        # thassa bout it
```
Comment (miningape) [inline src/pychart/_interpreter/pyparser.py:81]: Maybe move it somewhere else lol

File: src/pychart/_interpreter/statement.py [hunk 3]
```diff
@@ -81,3 +81,22 @@ def __call__(self, environment: Environment):
...
+        while_test: Expr,
+        while_body: Stmt,
+    ):
+        self.while_test = while_test
+        self.while_body = while_body
+
+    def __call__(self, environment: Environment):
+        while self.while_test(environment):
+            self.while_body(environment)
+        return None
```
Comment (miningape) [inline src/pychart/_interpreter/statement.py:102]: Just 1 suggestion for naming

```suggestion
class While(Stmt):
    
    test: Expr
    body: Stmt

    def __init__(
        self,
        test: Expr,
        body: Stmt,
    ):
        self.test = while_test
        self.body = while_body

    def __call__(self, environment: Environment):
        while self.test(environment):
            self.body(environment)
        return None
```

**Review by miningape — APPROVED** [review]
LGTM

File: src/pychart/__init__.py (hunk 1 above)
Comment (GangstersPairOfDice) [reply src/pychart/__init__.py:1]: Thank you.

File: src/pychart/_interpreter/pyparser.py (hunk 2 above)
Comment (GangstersPairOfDice) [reply src/pychart/_interpreter/pyparser.py:81]: Completely forgot about this hahahaha

**Review by GangstersPairOfDice — COMMENTED** [review]
Removed phd from py.parser comments

**Comment by GangstersPairOfDice** [comment]
Could someone verify the break statement please>

File: src/pychart/_interpreter/statement.py [hunk 4]
```diff
@@ -81,3 +81,34 @@ def __call__(self, environment: Environment):
...
             self.else_body(environment)
+
+
+class pychBreakStatementExceptionJeffer(Exception):
+    pass
+
+class Break(Stmt):
+    def __call__(self, environment: Environment):
+        raise pychBreakStatementExceptionJeffer("The jeeffer did not correspond")
+
```
Comment (miningape) [inline src/pychart/_interpreter/statement.py:92]: For readability and better error messages for the user

```suggestion
class BreakStatementException(Exception):
    pass

class Break(Stmt):
    def __call__(self, environment: Environment):
        raise BreakStatementException("Cannot invoke 'break;' outside of a while loop")

```

File: src/pychart/_interpreter/statement.py [hunk 5]
```diff
@@ -81,3 +81,34 @@ def __call__(self, environment: Environment):
...
+        body: Stmt,
+    ):
+        self.test = test
+        self.body = body
+
+    def __call__(self, environment: Environment):
+        while self.test(environment):
+            try:
+                self.body(environment)
+            except pychBreakStatementExceptionJeffer:
```
Comment (miningape) [inline src/pychart/_interpreter/statement.py:111]: See previous comment
```suggestion
            except BreakStatementException:
```

**Review by miningape — CHANGES_REQUESTED** [review]
Just changes for readability 

Do you also want to add continue? maybe a group decision

**Comment by spralja** [comment]
> Just changes for readability
> 
> Do you also want to add `continue`? maybe a group decision

ditto on `continue` one of the most underappreciated statements.

**Review by miningape — APPROVED** [review]

**Pull request merged**